update MQTT docs to include discover_ip option (#3809)
OttoWinter/esphomedocs: components/mqtt.rst

-  **retain** (*Optional*, boolean): If the published message should
   have a retain flag on or not. Defaults to ``true``.

.. _mqtt-device_discovery:

MQTT device discovery
---------------------

> The ESPHome device will respond to the following MQTT topics if `mqtt.discover_ip` is enabled.

Reviewer: All inline code should be double backticks, single just makes it italic.
```suggestion
The ESPHome device will respond to the following MQTT topics if ``mqtt.discover_ip`` is enabled.
```

Author: changed

- `esphome/discover` (All ESPHome device will answer)
- `esphome/ping/<APP_NAME>`

> The response will be send to `esphome/discover/<APP_NAME>` and is a JSON encoded message.

Reviewer 2: ```suggestion
The response will be sent to `esphome/discover/<APP_NAME>` and is a JSON encoded message.
```

The MQTT device discovery is currently used for:

- ESPHome dashboard (online / offline status)

> - ESPHome CLI (IP discovery e.g. for upload, logs)

Reviewer 2: ```suggestion
- ESPHome CLI (IP discovery; used to view logs and perform OTA uploads)
```

- Home Assistant device discovery

Example Payload:

.. code-block:: json

    {
      "ip": "192.168.0.122",
      "name": "esp32-test",
      "friendly_name": "Test Device",
      "port": 6053,
      "version": "2024.4.1",
      "mac": "84fce6123456",
      "platform": "ESP32",
      "board": "esp32-c3-devkitm-1",
      "network": "wifi",
      "api_encryption": "Noise_NNpsk0_25519_ChaChaPoly_SHA256"
    }

JSON keys:

-  **ip** (**Required**, ip): The IP address of the ESPHome device.

> -  **name** (**Required**, string): Name of the device (`esphome.name`).

Reviewer 2: ```suggestion
-  **name** (**Required**, string): Name of the device (for example: `my-device.local`).
```

Author: `esphome.name` is the name of the key in the yaml of the device, 
which is not coresponding to the mDNS name, so no `.local` will be added.

-  **mac** (**Required**, string): MAC address of the device.
-  **board** (**Required**, string): Board used for the device.
-  **version** (**Required**, string): ESPHome version.

> -  **port** (*Optional*, port): Port of the ESPHome API (if activated).

Reviewer 2: ```suggestion
-  **port** (*Optional*, port): Port of the ESPHome API (if enabled).
```

-  **ipX** (*Optional*, ip): Additional IP addresses (X is a number starting at 1).

> -  **friendly_name** (*Optional*, string): Friendly name of the device (`esphome.friendly_name`).

Reviewer 2: ```suggestion
-  **friendly_name** (*Optional*, string): Friendly name of the device (for example: `Bedroom lamp`).
```

Author: I provided example data in the JSON part, the idea with the `esphome.friendly_name` is to show from where the data is comming in the yaml definition of the device.
